tutteli-gradle-junitjacoco/src/main/groovy/ch/tutteli/gradle/junitjacoco/JunitJacocoPlugin.groovy

```diff
@@ -4,35 +4,32 @@ import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.logging.Logger
 import org.gradle.api.logging.Logging
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.tasks.testing.Test
 import org.gradle.testing.jacoco.plugins.JacocoPlugin
-import org.junit.platform.gradle.plugin.JUnitPlatformPlugin
 
 class JunitJacocoPlugin implements Plugin<Project> {
     private static final Logger LOGGER = Logging.getLogger(JunitJacocoPlugin.class)
 
-    static final String ARG_REPORTS_DIR = '--reports-dir'
-    static final String JUNIT_TASK_NAME = 'junitPlatformTest'
     static final String EXTENSION_NAME = 'junitjacoco'
-    static final String REPORT_TASK_NAME = 'junitPlatformJacocoReport'
+    static final String TEST_TASK_NAME = 'test'
+    static final String JACOCO_TASK_NAME = 'jacocoTestReport'
 
     @Override
     void apply(Project project) {
-        project.getPluginManager().apply(JUnitPlatformPlugin)
-        project.getPluginManager().apply(JacocoPlugin)
-        def junitPlatformTestTask = project.tasks.getByName(JUNIT_TASK_NAME)
-        def extension = project.extensions.create(EXTENSION_NAME, JunitJacocoPluginExtension, project, junitPlatformTestTask)
+        project.pluginManager.apply(JavaPlugin)
```

Comment:
is the java plugin really required?

Reply:
Yes, the `test` task is not present otherwise. Do you know a better way to have the `test` task created at this stage?

I do not want to have the `useJUnitPlatform()` call in `afterEvaluate`, because client code should be able to depend on the fact that junitPlatform is enabled.

Reply:
Fair enough, I don't see another solution and the java plugin is most likely enabled anyway (e.g. via Kotlin Plugin)